wagtail/wagtailadmin/views/pages.py

```diff
@@ -17,22 +17,38 @@
 from wagtail.wagtailadmin.forms import CopyForm, SearchForm
 from wagtail.wagtailadmin.utils import send_notification
 from wagtail.wagtailcore import hooks
-from wagtail.wagtailcore.models import Page, PageRevision, get_navigation_menu_items
+from wagtail.wagtailcore.models import (
+    Page, PageRevision, UserPagePermissionsProxy, get_explorable_page_paths, get_navigation_menu_items
+)
 
 
 def explorer_nav(request):
     return render(request, 'wagtailadmin/shared/explorer_nav.html', {
-        'nodes': get_navigation_menu_items(),
+        'nodes': get_navigation_menu_items(request.user),
     })
 
 
 def index(request, parent_page_id=None):
     if parent_page_id:
         parent_page = get_object_or_404(Page, id=parent_page_id)
+
+        if not UserPagePermissionsProxy(request.user).for_page(parent_page).can_explore():
+            if Page.objects.descendant_of(request.site.root_page).filter(pk=parent_page.id).exists():
+                # If the page is on the current Site, throw a 403: it's just an unpermitted page.
+                raise PermissionDenied
+            else:
+                # If the page isn't on the current site, throw a 404: the page doesn't exist on this Site.
+                raise Http404
     else:
+        # The user visited /admin/pages/, so we need to figure out the appropriate "root page" to show them.
         parent_page = Page.get_first_root_node()
+        if not request.user.is_superuser:
+            # Non-superusers get the Closest Common Ancestor of their permitted pages, if they have any.
+            permitted_pages, required_ancestors = get_explorable_page_paths(request.user)
+            if permitted_pages:
+                parent_page = Page.objects.get(path=required_ancestors[0])
 
-    pages = parent_page.get_children().prefetch_related('content_type')
+    pages = parent_page.get_explorable_children(request.user).prefetch_related('content_type')
```

Comment:
Are you sure that we can't just do the following?
```
pages = parent_page.get_children().prefetch_related('content_type')
```

As far as I know, it's not possible to have a subpage with less permissions than the parent.